Add a Contributing Guide

settermjd opened the pull request:
This PR:

- adds instructions showing how to contribute to the project.

:information_desk_person: As part of the overhaul of the documentation, I want to make it as easy as possible for others to know how to contribute to the project.

settermjd committed:
Add scripts section to composer.json

As part of adding CONTRIBUTING.md, the scripts section is required.
This is because it contains the scripts mentioned in CONTRIBUTING.md,
for running tests, PHPCS and so on.

settermjd committed:
Add a contributing guide

As part of the overhaul of the documentation, I want to make it as easy
as possible for others to know how to contribute to the project
— especially as it's an open source project.
This guide, taken from the Zend Expressive project, provides the
essentials on how to do so.

settermjd commented:
@shochdoerfer, this is still a work-in-progress, as there's a bit of extra information still required. But should be finished relatively quickly.

shochdoerfer commented:
Good starting point! For the ci tasks we have some phing tasks which you should refer to in the documentation:
 - To run CodeSniffer use: ./vendor/bin/phing sniff 
 - To run PHPUnit use: ./vendor/bin/phing unit 
 - To lint the PHP code use: ./vendor/bin/phing lint

Before sending a PR one should run `composer.phar validate` as the Travis build will also run the command and will fail if the composer.lock file is not up-to-date with the changes in the composer.json file.

settermjd committed:
Ignore Composer's vendor directory and PHPDocumentor's build directory

settermjd committed:
Add notes from Stephan regarding running composer validate

settermjd committed:
Updated the contributing guide

Added:

- the coding standard
- the IRC channel (and registered it too)
- added the private email address for security issues
- added the PGP key link

settermjd commented:
Ok, the initial draft's now done and it's ready for review.

settermjd committed:
Add a reference to the contributing guide in the readme

shochdoerfer commented on CONTRIBUTING.md:
```diff
@@ -0,0 +1,188 @@
+We request that you contact us via the email address above and give the project contributors a chance to resolve the vulnerability and issue a new release prior to any public exposure.
+This helps protect users and provides them with a chance to upgrade and/or update in order to protect their applications.
+
+For sensitive email communications, please use [our PGP key](http://pgp.mit.edu/pks/lookup?op=vindex&search=0xEC37D631657E0A89).
```
I generated a PGP key for the opensource@bitExpert.de email address, please use [this](http://pgp.mit.edu/pks/lookup?op=vindex&search=0x6DE7F8597CECDF13) link as reference.

shochdoerfer commented on CONTRIBUTING.md:
````diff
@@ -0,0 +1,188 @@
+2. Install dependencies via composer:
+
+```console
+$ curl -sS https://getcomposer.org/installer | php --
````
Not sure if it makes sense to better refer to the Composer documentation on how to [install Composer](https://getcomposer.org/download/) as their docs mention the more "secure" method.

settermjd replied:
Sounds good to me.

shochdoerfer commented on CONTRIBUTING.md:
````diff
@@ -0,0 +1,188 @@
+```
+### Sending a PR
+
+Before sending a PR, run `composer.phar validate`, as the Travis build will also run the command and will fail if the composer.lock file is not up-to-date with the changes in the composer.json file.
````
It should be mentioned that the CodeSniffer tests as well the PHPUnit tests need also to run.

settermjd replied:
Shall do. Thanks for the pointers while I'm still getting up to speed with the project.

shochdoerfer commented:
@settermjd Could you apply the few changes I mentioned above so that I can merge the PR?

settermjd committed:
Add changes suggested by Stephan

settermjd commented:
@shochdoerfer done!

shochdoerfer commented:
Thank you @settermjd for your first Disco PR!

settermjd commented:
Not a problem at all. Here's to many more!
